ocs_ci/ocs/couchbase.py

```diff
@@ -262,22 +264,42 @@ def get_couchbase_nodes(self):
             nodes_set.add(pod.get().get("spec").get("nodeName"))
         return list(nodes_set)
 
-    def teardown(self):
+    def cleanup(self):
         """
         Cleaning up the resources created during Couchbase deployment
 
         """
+        switch_to_project(constants.COUCHBASE_OPERATOR)
         if self.cb_create_cb_secret:
+            self.cb_secrets._is_deleted = False
             self.cb_secrets.delete()
         if self.cb_create_cb_cluster:
+            self.cb_example._is_deleted = False
             self.cb_example.delete()
         if self.cb_create_bucket:
+            self.cb_bucket._is_deleted = False
             self.cb_bucket.delete()
-        self.subscription_yaml.delete()
+        if self.cb_subscription:
+            self.subscription_yaml._is_deleted = False
+            self.subscription_yaml.delete()
         switch_to_project("default")
         self.ns_obj.delete_project(constants.COUCHBASE_OPERATOR)
         self.ns_obj.wait_for_delete(
             resource_name=constants.COUCHBASE_OPERATOR, timeout=90
         )
         PillowFight.cleanup(self)
         switch_to_default_rook_cluster_project()
+
+    def couchbase_full(self):
+        """
+        Run full CouchBase workload
+        """
+        # Create Couchbase subscription
+        self.couchbase_subscription()
+        # Create Couchbase worker secrets
+        self.create_cb_secrets()
+        # Create couchbase workers
+        self.create_cb_cluster(replicas=1)
+        self.create_data_buckets()
+        # Run couchbase workload
+        self.run_workload(replicas=1)
```

Comment:
do add this function in try .... except so that in-case of failures can fail with exceptions.

Reply:
Discussed offline with Ram, I am handling the failures case in the fixture teardown. No changes are required here.